llvm/lib/Analysis/FunctionPropertiesAnalysis.cpp

```diff
@@ -356,6 +369,40 @@ FunctionPropertiesUpdater::FunctionPropertiesUpdater(
     FPI.updateForBB(*BB, -1);
 }
 
+DominatorTree &FunctionPropertiesUpdater::getUpdatedDominatorTree(
+    FunctionAnalysisManager &FAM) const {
+  auto &DT =
+      FAM.getResult<DominatorTreeAnalysis>(const_cast<Function &>(Caller));
+
+  DenseSet<const BasicBlock *> NewSucc;
+  NewSucc.insert(succ_begin(&CallSiteBB), succ_end(&CallSiteBB));
+
+  // tell the DomTree about the new edges
+  std::deque<const BasicBlock *> Worklist;
+  Worklist.push_back(&CallSiteBB);
+  while (!Worklist.empty()) {
+    auto *BB = Worklist.front();
+    Worklist.pop_front();
+    assert(DT.getNode(BB));
+
+    for (auto *Succ : NewSucc) {
+      if (!DT.getNode(Succ))
+        Worklist.push_back(Succ);
+      DT.insertEdge(const_cast<BasicBlock *>(BB),
+                    const_cast<BasicBlock *>(Succ));
```

Comment:
Should collect updates here instead of calling insertEdge directly?

Reply:
I actually addressed this feedback incorrectly, and the original code was kind of nonsensical here (the working list was a bit... wat).

I believe cdd11d6 fixed it all, but please let me know if there are any concerns.